runtime/gc_base/RootScanner.cpp

```diff
@@ -930,10 +936,30 @@ MM_RootScanner::scanDoubleMappedObjects(MM_EnvironmentBase *env)
 				}
 			}
 		}
-		reportScanningEnded(RootScannerEntity_DoubleMappedObjects);
+		reportScanningEnded(RootScannerEntity_DoubleMappedOrVirtualLargeObjectHeapObjects);
 	}
 }
-#endif /* J9VM_GC_ENABLE_DOUBLE_MAP */
+#endif /* defined(J9VM_GC_ENABLE_DOUBLE_MAP) */
+
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+void
+MM_RootScanner::scanObjectsInVirtualLargeObjectHeap(MM_EnvironmentBase *env)
+{
+	if (_singleThread || J9MODRON_HANDLE_NEXT_WORK_UNIT(env)) {
+		GC_HeapRegionIteratorVLHGC regionIterator(_extensions->heap->getHeapRegionManager());
+		MM_HeapRegionDescriptorVLHGC *region = NULL;
+		reportScanningStarted(RootScannerEntity_DoubleMappedOrVirtualLargeObjectHeapObjects);
+		while (NULL != (region = regionIterator.nextRegion())) {
+			if (region->isArrayletLeaf()) {
+				J9Object *spineObject = (J9Object *)region->_allocateData.getSpine();
+				Assert_MM_true(NULL != spineObject);
+				doObjectInVirtualLargeObjectHeap(spineObject);
```

Comment:
we need to introduce a flag in region that we guarantie that we call this only once for each offheap object
doublemapping ensures this by setting arrayletDoublemapID->address only to first arraylet region